### PR TITLE
VM Clone validation - make sure all PVCs snapshot-able

### PR DESCRIPTION
**Pull request by mhenriks**
Signed-off-by: Michael Henriksen <mhenriks@redhat.com>

**What this PR does / why we need it**:

VM clone webhook should make sure all PVCs support snapshots.  Otherwise the clone target will refer to the same PVC as the source.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mhenriks** [comment]
cc @iholder-redhat @ShellyKa13

**Review by iholder-redhat — COMMENTED** [review]
Looks great!
left some small comments

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter.go [hunk 1]
```diff
@@ -239,5 +236,46 @@ func verifySourceExists(client kubecli.KubevirtClient, name, namespace string, s
 		}
 	}
 
-	return nil
+	// currently, cloning leverages vm snapshot/restore to copy volumes
+	// snapshot/restore requires that volumes support CSI snapshots
+	// this limitation should be removed eventually
+	// probably by leveraging CDI cloning
+	return verifyVolumesSupportSnapshot(vm, sourceField)
```
[the remark this answers is not in the document]
Comment (mhenriks) [reply pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter.go:243]: Refactored a bit to check kind and call explicit `validateCloneSourceVM` function.  There can be `validateCloneSourceXXX` functions added as we support different sources

**Comment by iholder-redhat** [comment]
/lgtm
Thanks @mhenriks! Looks great

**Pull request merged**